src/solver/impls/arkode/arkode_mri.hxx

```diff
@@ -0,0 +1,176 @@
+/**************************************************************************
+ * Interface to ARKODE MRI solver
+ * NOTE: ARKode is currently in beta testing so use with cautious optimism
+ *
+ * NOTE: Only one solver can currently be compiled in
+ *
+ **************************************************************************
+ * Copyright 2010-2024 BOUT++ contributors
+ *
+ * Contact: Ben Dudson, dudson2@llnl.gov
+ *
+ * This file is part of BOUT++.
+ *
+ * BOUT++ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * BOUT++ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with BOUT++.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **************************************************************************/
+
+#ifndef BOUT_ARKODE_MRI_SOLVER_H
+#define BOUT_ARKODE_MRI_SOLVER_H
+
+#include "bout/build_config.hxx"
+#include "bout/solver.hxx"
+
+#if not BOUT_HAS_ARKODE
```

Comment:
`SUNDIALS_VERSION_AT_LEAST(major, minor, patch)` is being added in #2990 so that could then be used as `#if SUNDIALS_VERSION_AT_LEAST(7,2,0)` if it will end up in 7.2.0